### PR TITLE
Actually add example templates

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

Whoops.  In https://github.com/flutter/flutter/pull/111276, I meant to include these templates for starting a sample.  The directory is linked from the README, but I forgot to add the files to the PR.

These are not meant to be linked from anywhere other than the README.md about creating samples.

## Related Issues
 - https://github.com/flutter/flutter/issues/96492

## Tests
 - added test templates too.

### CODE REVIEW

File: examples/api/lib/sample_templates/cupertino.0.dart [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// should be:
+//
+//   examples/api/lib/cupertino/baz/foo_bar.2.dart
+//
+// and it's associated test should be in:
+//
+//   examples/api/test/cupertino/baz/foo_bar.2_test.dart
+//
+// The following doc comment should remain, and be a doc comment so that the
+// symbol will be linked in the IDE. Don't put the whole description of the
```
Comment (HansMuller) [inline examples/api/lib/sample_templates/cupertino.0.dart:26]:
```suggestion
// symbol will be linked in the IDE. Don't use the whole description of the
```

File: examples/api/lib/sample_templates/cupertino.0.dart [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+// example, since that should already be in the API docs where this example is
+// referenced, and we don't want the two descriptions to diverge. If this sample
+// is referenced more than once, link back to the instance the example file is
+// named for.
+
+/// Flutter code sample for [Placeholder].
+
+import 'package:flutter/cupertino.dart';
+
+void main() => runApp(const SampleApp());
```
Comment (HansMuller) [inline examples/api/lib/sample_templates/cupertino.0.dart:36]: I wish we'd reserve `=>` for one-liners that return a value.

File: examples/api/lib/sample_templates/material.0.dart [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+// should be:
+//
+//   examples/api/lib/material/baz/foo_bar.2.dart
+//
+// and it's associated test should be in:
+//
+//   examples/api/test/material/baz/foo_bar.2_test.dart
+//
+// The following doc comment should remain, and be a doc comment so that the
+// symbol will be linked in the IDE. Don't put the whole description of the
```
Comment (HansMuller) [inline examples/api/lib/sample_templates/material.0.dart:26]:
```suggestion
// symbol will be linked in the IDE. Don't use the whole description of the
```

File: examples/api/lib/sample_templates/cupertino.0.dart [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+//
+//   examples/api/lib/<library>/<filename_without_dart>/<lower_snake_symbol>.<index>.dart
+//
+// So, if your example is the third example of the Foo.bar function in the
+// "baz.dart" file in the cupertino library, then the filename for your example
+// should be:
+//
+//   examples/api/lib/cupertino/baz/foo_bar.2.dart
+//
+// and it's associated test should be in:
```
Comment (HansMuller) [inline examples/api/lib/sample_templates/cupertino.0.dart:21]:
```suggestion
// and its associated test should be in:
```

File: examples/api/lib/sample_templates/material.0.dart [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+//
+//   examples/api/lib/<library>/<filename_without_dart>/<lower_snake_symbol>.<index>.dart
+//
+// So, if your example is the third example of the Foo.bar function in the
+// "baz.dart" file in the material library, then the filename for your example
+// should be:
+//
+//   examples/api/lib/material/baz/foo_bar.2.dart
+//
+// and it's associated test should be in:
```
Comment (HansMuller) [inline examples/api/lib/sample_templates/material.0.dart:21]:
```suggestion
// and its associated test should be in:
```

File: examples/api/lib/sample_templates/widgets.0.dart [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+//
+//   examples/api/lib/<library>/<filename_without_dart>/<lower_snake_symbol>.<index>.dart
+//
+// So, if your example is the third example of the Foo.bar function in the
+// "baz.dart" file in the widgets library, then the filename for your example
+// should be:
+//
+//   examples/api/lib/widgets/baz/foo_bar.2.dart
+//
+// and it's (required) associated test should be in:
```
Comment (HansMuller) [inline examples/api/lib/sample_templates/widgets.0.dart:21]:
```suggestion
// and its (required) associated test should be in:
```

File: examples/api/lib/sample_templates/widgets.0.dart [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+//
+//   examples/api/test/widgets/baz/foo_bar.2_test.dart
+//
+// The following doc comment should remain, and be a doc comment so that the
+// symbol will be linked in the IDE. Don't put the whole description of the
```
Comment (HansMuller) [inline examples/api/lib/sample_templates/widgets.0.dart:26]:
```suggestion
// symbol will be linked in the IDE. Don't use the whole description of the
```

**Review by HansMuller — APPROVED** [review]
LGTM

File: examples/api/lib/sample_templates/cupertino.0.dart (hunk 1 above)
Comment (gspencergoog) [reply examples/api/lib/sample_templates/cupertino.0.dart:26]: Done, in all three.

File: examples/api/lib/sample_templates/cupertino.0.dart (hunk 2 above)
Comment (gspencergoog) [reply examples/api/lib/sample_templates/cupertino.0.dart:36]: But it _does_, it returns `void`! :-) (I can tell you're unpersuaded: switched to brackets)

File: examples/api/lib/sample_templates/cupertino.0.dart (hunk 4 above)
Comment (gspencergoog) [reply examples/api/lib/sample_templates/cupertino.0.dart:21]: Oooh, that's a bad one, and I know better!

**Pull request merged**